Fwk/VizFwk/TestApps/Qt/QtMinimal_GLWidget/QMWidget_GLW.cpp

```diff
@@ -0,0 +1,215 @@
+
+
+
+//--------------------------------------------------------------------------------------------------
+/// 
+//--------------------------------------------------------------------------------------------------
+QMWidget_GLW::QMWidget_GLW(cvf::OpenGLContextGroup* contextGroup, QWidget* parent)
+:   cvfqt::GLWidget(contextGroup, QGLFormat(), parent)
+{
+    m_camera = new cvf::Camera;
+
+    ref<cvf::Rendering> rendering = new cvf::Rendering;
+    rendering->setCamera(m_camera.p());
+    rendering->addOverlayItem(new cvf::OverlayAxisCross(m_camera.p(), new cvf::FixedAtlasFont(cvf::FixedAtlasFont::STANDARD)));
+
+    m_renderSequence = new cvf::RenderSequence;
+    m_renderSequence->addRendering(rendering.p());
+
+    m_trackball = new cvf::ManipulatorTrackball;
+    m_trackball->setCamera(m_camera.p());
+}
+
+
+//--------------------------------------------------------------------------------------------------
+/// 
+//--------------------------------------------------------------------------------------------------
+QMWidget_GLW::~QMWidget_GLW()
+{
+}
+
+
+//--------------------------------------------------------------------------------------------------
+/// 
+//--------------------------------------------------------------------------------------------------
+void QMWidget_GLW::setScene(cvf::Scene* scene)
+{
+    ref<cvf::Rendering> rendering = m_renderSequence->firstRendering();
+    CVF_ASSERT(rendering.notNull());
+
+    rendering->setScene(scene);
+
+    if (scene)
+    {
+        cvf::BoundingBox bb = scene->boundingBox();
+        if (bb.isValid())
+        {
+            m_camera->fitView(bb, -cvf::Vec3d::Z_AXIS, cvf::Vec3d::Y_AXIS);
+            m_trackball->setRotationPoint(bb.center());
+        }
+    }
+
+    update();
+}
+
+
+//--------------------------------------------------------------------------------------------------
+///  
+//--------------------------------------------------------------------------------------------------
+void QMWidget_GLW::resizeGL(int width, int height)
+{
+    if (m_camera.notNull())
+    {
+        m_camera->viewport()->set(0, 0, width, height);
+        m_camera->setProjectionAsPerspective(m_camera->fieldOfViewYDeg(), m_camera->nearPlane(), m_camera->farPlane());
+    }
+}
+
+
+//--------------------------------------------------------------------------------------------------
+///  
+//--------------------------------------------------------------------------------------------------
+void QMWidget_GLW::paintEvent(QPaintEvent* /*event*/)
+{
+    QPainter painter(this);
+
+    makeCurrent();
+
+    cvf::OpenGLContext* currentOglContext = cvfOpenGLContext();
+    CVF_ASSERT(currentOglContext);
+    CVF_CHECK_OGL(currentOglContext);
+
+    painter.beginNativePainting();
+
+	cvf::OpenGLUtils::pushOpenGLState(currentOglContext);
+
+    if (m_renderSequence.notNull())
+    {
+        m_renderSequence->render(currentOglContext);
+    }
+
+	cvf::OpenGLUtils::popOpenGLState(currentOglContext);
+
```

Comment:
Funky indentation.

Reply:
tab vs space issues, will be fixed by clang-format